extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/main/java/io/quarkus/resteasy/reactive/links/deployment/LinksProcessor.java

```diff
@@ -194,6 +198,43 @@ private RuntimeValue<GetterAccessorsContainer> implementPathParameterValueGetter
         return getterAccessorsContainer;
     }
 
+    /**
+     * Validates if the given classname contains a field `id` or annotated with `@Id`
+     *
+     * @throws IllegalStateException if the classname does not contain any sort of field identifier
+     */
+    private void validateClassHasFieldId(IndexView index, String entityType) {
+        // create a new independent class name that we can override
+        DotName className = DotName.createSimple(entityType);
+        List<FieldInfo> classFields = index.getClassByName(className).fields();
+        List<AnnotationInstance> classAnnotations = index.getClassByName(className).annotations();
+
+        LinkedList<FieldInfo> allFields = new LinkedList<>(classFields);
+        LinkedList<AnnotationInstance> allAnnotations = new LinkedList<>(classAnnotations);
+
+        // also get fields and annotations from all super classes
+        DotName superClassName = index.getClassByName(className).superName();
+        while (superClassName != null) {
+            List<FieldInfo> allSuperFields = index.getClassByName(superClassName).fields();
```

Comment:
This `index.getClassByName(superClassName)` can return `null` if the superclass is not indexed, you should guard for that.

Reply:
@FroMage I added some more guards for the null cases and rewrote the code using recurrency, I think it really improves readability in this case